pkg/controller/istiocontrolplane/listeners.go

```diff
@@ -45,14 +45,19 @@ type IstioRenderingListener struct {
 	*helmreconciler.CompositeRenderingListener
 }
 
-var _ helmreconciler.RenderingListener = &IstioRenderingListener{}
-var _ helmreconciler.ReconcilerListener = &IstioRenderingListener{}
+// IstioStatusUpdater is a RenderingListener that updates the status field on the IstioControlPlane
+// instance based on the results of the Reconcile operation.
+type IstioStatusUpdater struct {
+	*helmreconciler.DefaultRenderingListener
+	instance   *v1alpha2.IstioControlPlane
+	reconciler *helmreconciler.HelmReconciler
+}
 
 // NewIstioRenderingListener returns a new IstioRenderingListener, which is a composite that includes IstioStatusUpdater
 // and IstioChartCustomizerListener.
-func NewIstioRenderingListener(instance *v1alpha1.IstioControlPlane) *IstioRenderingListener {
+func NewIstioRenderingListener(instance *v1alpha2.IstioControlPlane) *IstioRenderingListener {
```

Comment:
Actually there are several interfaces that either should be better documented to explain why they are needed, or removed. I'd vote for the second option since there's only one implementation right now and it's not clear when we would need to be able to handle multiple dimensions of implementations in a dynamically pluggable way - and for sure there's a price to pay for developers contributing to the code to ramp up on the complexity. Some abstractions that could potentially removed imo:
- factories like these: [1](https://github.com/istio/operator/blob/1d29ca219595b827c071d1e2506ca4a49ac3b925/pkg/controller/istiocontrolplane/customizer.go#L26), [2](https://github.com/istio/operator/blob/1d29ca219595b827c071d1e2506ca4a49ac3b925/pkg/controller/istiocontrolplane/customizer.go#L26), [3](https://github.com/istio/operator/blob/1d29ca219595b827c071d1e2506ca4a49ac3b925/pkg/helmreconciler/customizers.go#L98) and others - using these and passing them around everywhere makes code harder to follow and I'm not sure what it's buying us. 
- [RenderingInput](https://github.com/istio/operator/blob/1d29ca219595b827c071d1e2506ca4a49ac3b925/pkg/helmreconciler/interfaces.go#L72) - realistically do we need an abstraction for these? We already have an input definition from the CRD and it should cover everything we need. 

@rcernich did a great job create something that would be broad enough to accommodate future possibilities at a time when our direction was still not very clear. Now, we know a lot better what the first product will look like and I think devs will make faster progress if we roll back some of the complexity. We can certainly put these abstractions back if the need arises.